readthedocs_ext/_templates/readthedocs-insert.html.tmpl

```diff
@@ -1,21 +1,23 @@
 
 <!-- RTD Extra Head -->
 
-{%- if pagename == "index" %}
-    {%- set canonical_page = "" %}
-{%- elif pagename.endswith("/index") %}
-    {%- set canonical_page = pagename[:-("/index"|length)] + "/" %}
-{%- else %}
-    {%- set ending = "/" if builder == "readthedocsdirhtml" else ".html" %}
-    {%- set canonical_page = pagename + ending %}
+{%- if canonical_url %}
```

Comment:
Do we want to check for `and not html_baseurl`? Seems that would output 2x canonical URL still.

Reply:
We are doing that check in https://github.com/readthedocs/readthedocs.org/pull/7540.
We could do the check here again, but seems redundant

Reply:
we could also do all the checks here, but by doing it in .org we can set the value in `html_baseurl` and rely on sphinx itself to set the canonical url instead of us doing it.

Reply:
Gotcha, makes sense.